src/storage/record_descriptor.cpp

```diff
@@ -321,9 +321,12 @@ record_descriptor::unpack (cubpacking::unpacker &unpacker)
 {
   unpacker.unpack_short (m_recdes.type);
   unpacker.peek_unpack_buffer_length (m_recdes.length);
+
+  // resize_buffer requires m_data_source to be set
+  m_data_source = data_source::COPIED;
```

Comment:
can it be set from the beginning?

Reply:
Yes it can be :) I chose to set it explicitly right before the operation that requires it